### PR TITLE
Pavan/branch aug 18

### PR DESCRIPTION
**Pull request by PavanKumarDega**
Empployee and Signup Creation

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -14,16 +14,17 @@ gem 'puma', '~> 5.0'
 # Use Redis adapter to run Action Cable in production
 # gem 'redis', '~> 4.0'
 # Use Active Model has_secure_password
-# gem 'bcrypt', '~> 3.1.7'
-
+gem 'bcrypt', '~> 3.1.7'
+gem 'jwt'
```
Comment (srihari-maddineni) [inline Gemfile:18]: @PavanKumarDega are you using this gem. if not, remove it.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,2 +1,18 @@
...
+ #   header = header.split(' ').last if header
+ #   begin
+ #     @decoded = JsonWebToken.decode(header)
+ #     @current_user = User.find(@decoded[:user_id])
+ #   rescue ActiveRecord::RecordNotFound => e
+ #     render json: { errors: e.message }, status: :unauthorized
+ #   rescue JWT::DecodeError => e
+ #     render json: { errors: e.message }, status: :unauthorized
+ #   end
+ # end
```
Comment (srihari-maddineni) [inline app/controllers/application_controller.rb:17]: @PavanKumarDega why you want to merge the code which is commented.

File: app/controllers/concerns/json_web_token.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class JsonWebToken
+  SECRET_KEY = Rails.application.secrets.secret_key_base. to_s
```
Comment (srihari-maddineni) [inline app/controllers/concerns/json_web_token.rb:2]: @PavanKumarDega remove this code if not using

File: app/controllers/employees_controller.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  @employee.destroy
+end
+
+private
+  def set_employee
+    @employee = Employee.find(params[:id])
+  end
+
+  def employee_params
+    params.require(:employee).permit(:name,:email,:phonenumber, :emp_id, :devices, :role)
```
Comment (srihari-maddineni) [inline app/controllers/employees_controller.rb:52]: - param :devices required??
- :phonenumber should be :phone_number
- :role should be roles

File: app/controllers/employees_controller.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
+class EmployeesController < ApplicationController
+  before_action :set_employee, only: [:show, :update, :destroy]
+  #after_action :user_signup, only: [:create]
```
Comment (srihari-maddineni) [inline app/controllers/employees_controller.rb:3]: don't push the code which is commented

File: app/controllers/employees_controller.rb [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  @employee = Employee.new
+
+  render json: @employee, serializer: EmployeeSerializer
+end
+def show
+  render json: @employee, serializer: EmployeeSerializer
+end
+
+def create
+  @employee = Employee.new(employee_params)
```
Comment (srihari-maddineni) [inline app/controllers/employees_controller.rb:20]: - why can't you use Employee.create()
- and def new can be removed

File: app/controllers/employees_controller.rb [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  if @employee.save
+    @user = User.new(:employee_id =>@employee_id, :email => @employee.email, :password => @employee.emp_id)
```
Comment (srihari-maddineni) [inline app/controllers/employees_controller.rb:22]: use callback to create the user, after employee creation is successful

File: app/controllers/employees_controller.rb [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+def update
+  if @employee.update(employee_params)
+    render json: @employee, serializer: EmployeeSerializer
+  else
+    render json: @employee.errors, status: :unprocessable_entity
+  end
+end
+
+def destroy
+  @employee.destroy
```
Comment (srihari-maddineni) [inline app/controllers/employees_controller.rb:43]: return head :no_content, after the employee is destroyed

File: app/controllers/users_controller.rb [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+  def destroy
+    @user.destroy
+  end
+
+  private
+
+  def find_user
+    @user = User.find_by(params[:id])
+    rescue ActiveRecord::RecordNotFound
+      render json: { errors: 'User not found' }, status: :not_found
```
Comment (srihari-maddineni) [inline app/controllers/users_controller.rb:29]: why you want to render something here, is it an action method which is supposed to return something?

File: test/controllers/authentication_controller_test.rb [hunk 10]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class AuthenticationControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
```
Comment (srihari-maddineni) [inline test/controllers/authentication_controller_test.rb:4]: empty tests?
either you implement the test cases or remove them from the pr

**Review by srihari-maddineni — CHANGES_REQUESTED** [review]

**Comment by PavanKumarDega** [comment]
i need to make some changes to my code

**Pull request closed without merging**